api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -24,11 +22,12 @@
     public static final String INVALID = "<%s: Invalid>";
     public static final String NOT_FOUND = "<%s: Not Found>";
 
-    public static final Pattern NUMERIC_PATTERN = Pattern.compile("[1-9][0-9]*|0");
-    public static final Pattern NUMERIC_PATTERN_10 = Pattern.compile("[1-9][0-9]{9}");
-    public static final Pattern NUMERIC_PATTERN_15 = Pattern.compile("[1-9][0-9]{14}");
+    public static final long SMALLEST_10_DIGIT_NUMBER = 1000000000L;
+    public static final long LARGEST_10_DIGIT_NUMBER  = 9999999999L;
+    public static final long SMALLEST_15_DIGIT_NUMBER = 100000000000000L;
+    public static final long LARGEST_15_DIGIT_NUMBER  = 999999999999999L;
 
-    protected static boolean validateFieldPresent(StringBuilder errors, String fieldName, String value) {
+    protected static boolean validateFieldPresent(StringBuilder errors, String fieldName, Object value) {
```

Comment:
Technically we are validating the field has a value.  I don't think we can actually determine the difference between these two json objects:

```javascript
{
   "bar": 1,
   "foo": null
}
```
and
```javascript
{
   "bar": 1
}
```

So I'm not sure we can every return a truly correct error.  I'm also not sure if they will get confused if we say it's not present when really it has no value.